### PR TITLE
Feature/registration codes

### PR DESCRIPTION
**Pull request by MTaylor-tech**
The backend groundwork for a registration code system. Whenever a company is created, default roles are added to the company. Whenever a role is created, a random 6 character code is created. That code can be sent by the admin to their users. When entered, it can be used to associate the user with a company and role.

A company's roles can be checked at `/companies/{companyId}/roles`
A code can be checked at `/companies/roles/{code}`

### CODE REVIEW

**Comment by MTaylor-tech** [comment]
[As an Admin, I'd like to get a special URL with a code to register my employees and clients](https://trello.com/c/TVgkHbiI/21-as-an-admin-id-like-to-get-a-special-url-with-a-code-to-register-my-employees-and-clients)

File: helpers/genRandomCodes.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+const alpha = [
```
Comment (Cireimu) [inline helpers/genRandomCodes.js:3]: Could we maybe use something like nanoid instead to generate the codes?

**Review by Cireimu — APPROVED** [review]
Looks good overall, just might want to go back over the genRandomCodes.js but overall still seems to perform the expected functionality.

File: helpers/genRandomCodes.js (hunk 1 above)
Comment (MTaylor-tech) [reply helpers/genRandomCodes.js:3]: Two reasons we aren't:
1) I didn't know about nanoid and for some reason didn't think to look for libraries that did this.
2) Our use-case is really simple and isn't being used in a high-security situation, so I think this is good, at least for now.
I did check out nanoid, but it seemed overly complicated for what we need. If we need something more secure in the future, I will look at it again and compare it to similar libraries.

**Comment by JC8747** [comment]
Everything looks good. Great work as usual on the backend portion of this. The duplication check is a good idea that would have been easy to miss and could have caused a lot of problems later down the line.

**Review by JC8747 — APPROVED** [review]
Everything looks good. Great work as usual on the backend portion of this. The duplication check is a good idea that would have been easy to miss and could have caused a lot of problems later down the line.

**Pull request merged**